bundles/org.dataflowanalysis.analysis.pcm.informationflow/src/org/dataflowanalysis/analysis/pcm/informationflow/core/extraction/IFPCMExtractionStrategy.java

```diff
@@ -0,0 +1,317 @@
+		Map<String, List<T>> variableNameToVarChars = new HashMap<>();
+		for (var varChar : varChars) {
+			String variableName = varChar.getVariableUsage_VariableCharacterisation().getNamedReference__VariableUsage()
+					.getReferenceName();
+			var mappedVarChars = variableNameToVarChars.get(variableName);
+
+			if (mappedVarChars == null) {
+				mappedVarChars = new ArrayList<>();
+			}
+			mappedVarChars.add(varChar);
+			variableNameToVarChars.put(variableName, mappedVarChars);
+		}
+		return variableNameToVarChars;
+	}
+
+	private List<AbstractNamedReference> extractVariableDependencies(
+			List<VariableCharacterisation> variableCharacterisations) {
+		return variableCharacterisations.stream()
+				.flatMap(varChar -> IFStoexUtils
+						.findVariablesInExpression(varChar.getSpecification_VariableCharacterisation().getExpression())
+						.stream().map(dependency -> dependency.getId_Variable()))
+				.toList();
+	}
+
+	private List<ConfidentialityVariableCharacterisation> calculateConfidentialityVariableCharacteristationsFromReferences(
+			List<AbstractNamedReference> references, AbstractNamedReference characterisedVariable) {
+
+		// Set the lowest level if the VariableCharacterisation contains no variables
+		if (references.size() <= 0) {
+			return IFConfidentialityVariableCharacterisationUtils.createSetLowestLevelCharacterisationsForLattice(
+					characterisedVariable, getLatticeCharacteristicType(), getLattice());
+		}
+
+		return IFConfidentialityVariableCharacterisationUtils.createMaximumJoinCharacterisationsForLattice(
+				characterisedVariable, references, getLatticeCharacteristicType(), getLattice());
+	}
+
+	/**
+	 * Determines the resulting effective
+	 * {@link ConfidentialityVariableCharacterisation} from calculated
+	 * characterizations, defined characterizations and optionally an security
+	 * context.
+	 * 
+	 * @param calculatedCharacterisations the calculated characterizations
+	 * @param definedCharacterisations    the defined characterizations
+	 * @param optionalSecurityContext     the optional security context
+	 * @return the resulting characterizations
+	 */
+	protected abstract List<ConfidentialityVariableCharacterisation> calculateResultingConfidentialityVaraibleCharacterisations(
+			List<ConfidentialityVariableCharacterisation> calculatedCharacterisations,
+			List<ConfidentialityVariableCharacterisation> definedCharacterisations,
+			Optional<DataFlowVariable> optionalSecurityContext);
+
+	/**
+	 * Returns the lattice used for the extraction.
+	 * 
+	 * @return the lattice used for the extraction
+	 */
+	protected Enumeration getLattice() {
+		if (!initialized) {
+			initializeResources();
+		}
+		return lattice;
+	}
+
+	/**
+	 * Returns the CharacteristicType used for the extraction.
+	 * 
+	 * @return the CharacteristicType used for the extraction
+	 */
+	protected CharacteristicType getLatticeCharacteristicType() {
+		if (!initialized) {
+			initializeResources();
+		}
+		return latticeCharacteristicType;
```

Comment:
You could convert this to a one-liner with a ternary operator (`boolean value ? true : false`)

Reply:
As far as I understand the ternary operator, it behaves like an if-else block. Since there is an optional initialization step and an return which should happen afterward, I don't understand how the ternery operator can shorten the code.